framework/quota/src/org/apache/cloudstack/quota/constant/QuotaTypes.java

```diff
@@ -0,0 +1,99 @@
+        quotaTypeList.put(SNAPSHOT, new QuotaTypes(SNAPSHOT, "SNAPSHOT", "GB-Month", "Snapshot Usage"));
+        quotaTypeList.put(SECURITY_GROUP, new QuotaTypes(SECURITY_GROUP, "SECURITY_GROUP", "Policy-Month", "Security Group Usage"));
+        quotaTypeList.put(LOAD_BALANCER_POLICY, new QuotaTypes(LOAD_BALANCER_POLICY, "LOAD_BALANCER_POLICY", "Policy-Month", "Load Balancer Usage"));
+        quotaTypeList.put(PORT_FORWARDING_RULE, new QuotaTypes(PORT_FORWARDING_RULE, "PORT_FORWARDING_RULE", "Policy-Month", "Port Forwarding Usage"));
+        quotaTypeList.put(NETWORK_OFFERING, new QuotaTypes(NETWORK_OFFERING, "NETWORK_OFFERING", "Policy-Month", "Network Offering Usage"));
+        quotaTypeList.put(VPN_USERS, new QuotaTypes(VPN_USERS, "VPN_USERS", "Policy-Month", "VPN users usage"));
+        quotaTypeList.put(VM_DISK_IO_READ, new QuotaTypes(VM_DISK_IO_READ, "VM_DISK_IO_READ", "GB", "VM Disk usage(I/O Read)"));
+        quotaTypeList.put(VM_DISK_IO_WRITE, new QuotaTypes(VM_DISK_IO_WRITE, "VM_DISK_IO_WRITE", "GB", "VM Disk usage(I/O Write)"));
+        quotaTypeList.put(VM_DISK_BYTES_READ, new QuotaTypes(VM_DISK_BYTES_READ, "VM_DISK_BYTES_READ", "GB", "VM Disk usage(Bytes Read)"));
+        quotaTypeList.put(VM_DISK_BYTES_WRITE, new QuotaTypes(VM_DISK_BYTES_WRITE, "VPN_USERS", "GB", "VM Disk usage(Bytes Write)"));
+        quotaTypeList.put(VM_SNAPSHOT, new QuotaTypes(VM_SNAPSHOT, "VM_SNAPSHOT", "GB-Month", "VM Snapshot storage usage"));
+        quotaTypeList.put(CPU_CLOCK_RATE, new QuotaTypes(CPU_CLOCK_RATE, "CPU_CLOCK_RATE", "Compute-Month", "Quota tariff for using 1 CPU of clock rate 100MHz"));
+        quotaTypeList.put(CPU_NUMBER, new QuotaTypes(CPU_NUMBER, "CPU_NUMBER", "Compute-Month", "Quota tariff for running VM that has 1vCPU"));
+        quotaTypeList.put(MEMORY, new QuotaTypes(MEMORY, "MEMORY", "Compute-Month", "Quota tariff for using 1MB or RAM for 1 hour"));
+    }
+
+    private QuotaTypes(Integer quotaType, String name, String unit, String description) {
+        this.quotaType = quotaType;
+        this.description = description;
+        this.quotaName = name;
+        this.quotaUnit = unit;
+        this.discriminator = "None";
+    }
+
+    public static HashMap<Integer, QuotaTypes> listQuotaTypes() {
+        return quotaTypeList;
+    }
+
+    public String getDiscriminator() {
+        return discriminator;
+    }
+
+    public String getQuotaName() {
+        return quotaName;
+    }
+
+    public String getQuotaUnit() {
+        return quotaUnit;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public Integer getQuotaType() {
+        return quotaType;
+    }
+
+    static public String getDescription(int quotaType) {
+        HashMap<Integer, QuotaTypes> quotaMap = listQuotaTypes();
+        if (quotaMap.containsKey(quotaType)) {
+            return quotaMap.get(quotaType).getDescription();
+        }
+        return null;
```

Comment:
I think if check is to avoid NPE when we call .getDescription() over map entry.
May be we can make it 
QuotaTypes qTypes = quotaMap.get(quotaType);
if (qTypes != null) {
    return qTypes.getDescription();
}